tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -137,6 +137,11 @@ sealed trait NativeConfig {
     }
   }
 
+  /** Shall enable the use of C++ runtime. Useful if you want to compile c++
+   *  code in your Scala Native application.
+   */
+  def useCPPRuntime: Boolean
```

Comment:
I'm not really a fun of this setting to be honest.

At first it might be misleading - actually it's only controlling which program should we use for linking `clang` or `clang++`. If so, then maybe we should introduce `alternativeLinker: Option[Path]` which would be to used if defined, altough it might be problematic in usage becouse of hard-coded settings we enforce, so probably that's not a good idea to follow.

From the user perspective I'd prefer to not need any additional configuration, only because upstream library I use is using .cpp files for the internals. So we should have some way of automatic detection when c++ stdlib is really needed.  
One idea how to achive that would be checking if we're compiling any .cpp files if so we'd use clang++ to link and clang otherwise. There are 2 issues however. First problem is we're still defining .cpp sources in core Scala Native project - we ship libunwind which contains C++ sources, we can consider removing it in the follow up PR, a quick experiment on my machine proved there is no issues with that.  
The second problem is that we can find effectively empty .cpp files becouse their content can be erased by pre-processor - in fact that's what we frequently use. We could potentially inspect output `.o` files with `llvm-nm` to check if it contains at least 1 symbol. Still, we cannot guarantee this binary would be reachable, so it maybe not be reliable. 

From what I've observed the only difference between using `clang` vs `clang++` for linking `.o` files is addition of `-lc++`. Both are delegating to `ld` under the hood using the same options. 
Based on that we can take one more direction - let's put the responsability on the author of the glue code using c++. As an author of library you can either put `@link("c++")` or create `resources/scala-native.properties` files with `nir.link.names=c++` to ensure the c++ stdlib is linked. 
We can take advantage of this fact even further and detect if `LLVM.prepareLinkCommand -> links variable` contains `c++` entry and use it to control if `clang` or `clang++` should be used. 

I think that the last solution might be the most reliable in the long term. We would be able to combine our about  using any c++ features internally by toolchain (.eg`-fcxx-exceptions`, targeting Windows) with source metadata from users (`@link`). 
If going with this solutions we should probably add documentation section how user/library author can enable c++ support.

Reply:
> There are 2 issues however. First problem is we're still defining .cpp sources in core Scala Native project - we ship libunwind which contains C++ sources, we can consider removing it in the follow up PR, a quick experiment on my machine proved there is no issues with that.

I'm not sure I understand what is the problem with libunwind. libunwind itself uses the `-fno-rtti -fno-exceptions -funwind-tables` flags to compile its c++ with not reliance on the c++ runtime.

Reply:
About implicitly adding libc++ based on having a single c++ seems too intrusive. I would prefer a manual way if possible. For sure I like that `clang++` can be replaced by the `lc++` flag, I think I can do that already, which would simplify the discussion.